tools/pylib/boutdata/collect.py

```diff
@@ -141,63 +203,73 @@ def getDataFile(i):
         except KeyError:
             myg = 0
             print("MYG not found, setting to {}".format(myg))
-            
-        if tind is not None or xind is not None or yind is not None or zind is not None:
-            raise ValueError("tind, xind, yind, zind arguments are not implemented yet for single (parallel) data file")
-        
+
         if xguards:
-            xstart = 0
-            xlim = None
+            nx = f["nx"]
         else:
-            xstart = mxg
-            if mxg > 0:
-                xlim = -mxg
-            else:
-                xlim = None
+            nx = f["nx"] - 2*mxg
```

Comment:
For this single data file case, I think xguards==False needs fixing. In the multiple file case below, the boundary guard cells get cut out if xguards==False before xind is used. But here, xind is just passed straight to `DataFile.read()` so to maintain the same behaviour as the multiple-file case, we need to offset the start and stop values of xind by mxg if xguards==False.

Reply:
Could you email me a failing example so I get this right? Unfortunately, we're missing a test suite for the python tools